### DISCUSSION TITLE
LemonLDAP configuration

### DISCUSSION THREAD

**Discussion by tomflenner** [nfrastack/container-lemonldap#23] [Q&A] [answered]
Hello !

I have a question about the LemonLDAP Configuration, on this docker image we have two env variables :

- SETUP_TYPE [Default =Auto] : The other value accepted is Manual ?
- CONFIG_TYPE = File, Rest

If i run my container as SETUP_TYPE = MANUAL, does the image care about CONFIG_TYPE value ? I want to put this container configuration as backend database

Thanks !

**Comment by tiredofit** [comment, marked as answer]
Here's what I would do: 
Use `SETUP_TYPE=AUTO` for the first run, have it create a `lemonldap-ng.ini` file for you.
Then switch it over to `SETUP_TYPE=MANUAL` and you can then modify the `lemonldap-ng.ini` file and your local changes will persist on container restarts to change your configuration backend. I used to have SQL and LDAP backends built into the Auto mode but supporting was a pain.

If you were looking at putting out more handlers, you would want to set your Global Sessions and Config in your `lemonldap-ng.ini` file on your master server, and then you could use `CONFIG_TYPE=REST` on your remote handlers to connect to your portal with `SETUP_TYPE=AUTO`. You'd need to setup REST inside your manager site to support config and session sharing. 

There are a bunch of combinations you could do, but I built the AUTO mode to get someone up and running with sane defaults _AND_ to quickly be able to deploy new handlers, specifically in the environment in how I operate.
